File: javascript/01_AboutStrings/checkString_02.js


// 1. ฟังก์ชันตรวจสอบข้อความ - ใช้สำหรับวิเคราะห์คุณสมบัติต่างๆ ของข้อความที่รับเข้ามา
function checkString(str) {
    // ตรวจสอบว่าข้อความว่างหรือเป็น null หรือไม่
    if (!str || str.length === 0) {
        return "ข้อความว่าง";
    }
    
    // ใช้ Regular Expression ตรวจสอบว่ามีตัวเลขในข้อความหรือไม่
    const hasNumbers = /\d/.test(str);
    
    // ตรวจสอบว่ามีตัวอักษรพิเศษหรือไม่
    const hasSpecialChars = /[!@#$%^&*(),.?":{}|<>]/.test(str);
    
    // ส่งคืนออบเจ็กต์ที่มีข้อมูลการวิเคราะห์
    return {
        length: str.length,                        // ความยาวของข้อความ
        hasNumbers: hasNumbers,                    // มีตัวเลขหรือไม่
        hasSpecialChars: hasSpecialChars,         // มีตัวอักษรพิเศษหรือไม่
        isUpperCase: str === str.toUpperCase(),   // เป็นตัวพิมพ์ใหญ่ทั้งหมดหรือไม่
        isLowerCase: str === str.toLowerCase()    // เป็นตัวพิมพ์เล็กทั้งหมดหรือไม่
    };
}

console.log(checkString("Hello123!"));
console.log(checkString("HELLO"));
console.log(checkString("1204"));

File: java/01_AboutStrings/checkString_02.java
public class checkString_02 {
    // 1. คลาสสำหรับเก็บผลการวิเคราะห์ข้อความ
    public static class StringAnalysis {
        public int length;
        public boolean hasNumbers;
        public boolean hasSpecialChars;
        public boolean isUpperCase;
        public boolean isLowerCase;

        @Override
        public String toString() {
            return "StringAnalysis{" +
                   "length=" + length +
                   ", hasNumbers=" + hasNumbers +
                   ", hasSpecialChars=" + hasSpecialChars +
                   ", isUpperCase=" + isUpperCase +
                   ", isLowerCase=" + isLowerCase +
                   '}';
        }
    }

    public static StringAnalysis checkString_02(String input) {
        StringAnalysis analysis = new StringAnalysis();
        analysis.length = input.length();
        analysis.hasNumbers = input.matches(".*\\d.*");
        analysis.hasSpecialChars = input.matches(".*[^a-zA-Z0-9].*");
        analysis.isUpperCase = input.equals(input.toUpperCase());
        analysis.isLowerCase = input.equals(input.toLowerCase());
        return analysis;
    }

    public static void main(String[] args) {
        // ทดสอบการวิเคราะห์ข้อความ
        System.out.println(checkString_02("Hello123!"));
    }
}